cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -255,27 +265,59 @@ runQueryTip (AnyConsensusModeParams cModeParams) network mOutFile = do
       Aeson.Object $ obj <> HMS.fromList [name .= Aeson.Null]
     toObject _ _ _ = Aeson.Null
 
-queryEraHistoryAndSystemStart
+queryChainTipAndEraHistoryAndSystemStart
   :: LocalNodeConnectInfo CardanoMode
   -> Maybe ChainPoint
-  -> IO (Either Net.Query.AcquireFailure (EraHistory CardanoMode, SystemStart))
-queryEraHistoryAndSystemStart connctInfo mpoint = do
-    resultVar <- newEmptyTMVarIO
+  -> IO (ChainTip, Either Net.Query.AcquireFailure (EraHistory CardanoMode, SystemStart))
+queryChainTipAndEraHistoryAndSystemStart connctInfo mpoint = do
+    resultVarEraHistoryAndSystemStart <- newEmptyTMVarIO
+    resultVarChainTip <- newEmptyTMVarIO
+
+    waitResult <- pure $ (,)
+      <$> readTMVar resultVarChainTip
+      <*> readTMVar resultVarEraHistoryAndSystemStart
+
     connectToLocalNode
       connctInfo
       LocalNodeClientProtocols
-      { localChainSyncClient    = NoLocalChainSyncClient
-      , localStateQueryClient   = Just (singleQuery mpoint resultVar)
+      { localChainSyncClient    = LocalChainSyncClient $ chainSyncGetCurrentTip waitResult resultVarChainTip
+      , localStateQueryClient   = Just (singleQuery waitResult mpoint resultVarEraHistoryAndSystemStart)
       , localTxSubmissionClient = Nothing
       }
-    atomically (takeTMVar resultVar)
+
+    atomically waitResult
   where
+    chainSyncGetCurrentTip
+      :: forall mode a
+      .  STM a
```

Comment:
You can ignore this suggestion:
```suggestion
      :: forall mode
      .  (forall a. STM a)
```
I think this will encode that you don't need the value at all.

Reply:
I like this suggestion.  Thanks!

Reply:
This is odd.  The suggested change doesn't compile:

```

src/Cardano/CLI/Shelley/Run/Query.hs:283:81: error:
    • Couldn't match type ‘a’
                     with ‘(ChainTip,
                            Either AcquireFailure (EraHistory CardanoMode, SystemStart))’
      ‘a’ is a rigid type variable bound by
        a type expected by the context:
          forall a. STM a
        at src/Cardano/CLI/Shelley/Run/Query.hs:283:81-90
      Expected type: STM a
        Actual type: STM
                       (ChainTip,
                        Either AcquireFailure (EraHistory CardanoMode, SystemStart))
    • In the first argument of ‘chainSyncGetCurrentTip’, namely
        ‘waitResult’
      In the second argument of ‘($)’, namely
        ‘chainSyncGetCurrentTip waitResult resultVarChainTip’
      In the ‘localChainSyncClient’ field of a record
    |
283 |       { localChainSyncClient    = LocalChainSyncClient $ chainSyncGetCurrentTip waitResult resultVarChainTip
```

Reply:
I'll keep the code as it is for now.